client/my-sites/checkout/upsell-nudge/purchase-modal/index.tsx

```diff
@@ -109,6 +125,44 @@ export default function PurchaseModalWrapper( props: PurchaseModalProps ) {
 		]
 	);
 
+	useEffect( () => {
+		let isUpdatingCart = false;
+
+		if ( storedCard && countries?.length && ! isUpdatingCart ) {
+			const vatDetails: VatDetails = {
+				country: storedCard.tax_location?.country_code,
+				id: storedCard.tax_location?.vat_id,
+				name: storedCard.tax_location?.organization,
+				address: storedCard.tax_location?.address,
+			};
+			const contactInfo: ManagedContactDetails = {
+				state: wrapValueInManagedValue( storedCard.tax_location?.subdivision_code ),
+				city: wrapValueInManagedValue( storedCard.tax_location?.city ),
+				postalCode: wrapValueInManagedValue( storedCard.tax_location?.postal_code ),
+			};
+
+			updateCartContactDetailsForCheckout(
+				countries ?? [],
+				responseCart,
+				updateLocation,
+				contactInfo,
+				vatDetails
+			);
+			replaceProductsInCart( [ productToAdd ] );
+		}
+
+		return () => {
+			isUpdatingCart = true;
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ replaceProductsInCart, updateLocation, storedCard, productToAdd, countries ] );
+
+	const handleOnClose = () => {
+		Promise.all( [ updateLocation( { countryCode: '' } ), replaceProductsInCart( [] ) ] ).catch();
```

Comment:
`Non-blocking:`

I see that this was existing logic, but it's really curious that we reset the country code as well. I wonder why 🤔

Reply:
Good question! It looks like I added it in https://github.com/Automattic/wp-calypso/pull/74555, specifically this commit: https://github.com/Automattic/wp-calypso/pull/74555/commits/202de1533f43b553d9e433748431b204e2244d87. I don't remember the context, but presumably there was a good reason if I made a whole commit for it. It appears to have something to do with sending the correct tax location so it may have to do with forcing the location to be repopulated correctly for a different card. Because there's no tax form shown in the modal, the information has to come from the card only, whereas normally the tax form would override the modal, and if the value is already set by a previous card, it might cause problems. That's a guess, though.